pkg/transformer/transformer.go

```diff
@@ -20,12 +20,20 @@ type NodeTerm struct {
 	Annotations      map[string]string               `yaml:"annotations,omitempty"`
 	Labels           map[string]string               `yaml:"labels,omitempty"`
 	PlatformMetadata map[string]string               `yaml:"platformMetadata,omitempty"`
+	Features         NodeFeaturesFlagSpec            `yaml:"features,omitempty"`
 }
 
 // NodeSpec represents the transformed node specifcations.
 type NodeSpec struct {
 	Annotations map[string]string
 	Labels      map[string]string
+	Features    NodeFeaturesFlagSpec
+}
+
+// NodeFeaturesFlagSpec represents the node features flags.
+type NodeFeaturesFlagSpec struct {
+	// PublicIPDiscovery try to find public IP on the node
+	PublicIPDiscovery *bool `yaml:"publicIPDiscovery,omitempty"`
```

Comment:
we could just make this non-pointer and simplify the check

Reply:
I thought it will use as true/false/undefined value to set default in future.
Currently we have `false` as default. And i've changed it to non pointer.

Thanks.